src/random_lcg.cpp

```diff
@@ -28,47 +27,41 @@ constexpr uint64_t prn_stride {152917LL};                // stride between
                                                          //   particles
 constexpr double   prn_norm   {1.0 / prn_mod};           // 2^-63
 
-// Current PRNG state
-uint64_t prn_seed[N_STREAMS];  // current seed
-int      stream;               // current RNG stream
-#pragma omp threadprivate(prn_seed, stream)
-
-
 //==============================================================================
 // PRN
 //==============================================================================
 
 extern "C" double
-prn()
+prn(uint64_t* prn_seeds, int stream)
```

Comment:
I was thinking about this a little bit and came up with two possibilities for improving this:

1. Create a struct, something like `struct prn_state { uint64_t* seeds; int stream }` that could be passed as a single argument through all the function calls rather than having to pass seeds/stream separately.
2. Since `prn` only manipulates a single stream, there's actually no good reason to be passing the stream explicitly. Instead, we can just have it accept `uint64_t& seed` (assuming we can remove the `extern "C"`) and it would be the responsibility of the caller to send in the seed corresponding to the correct stream.

I like the simplicity of option 2, so I'm leaning towards that but would like to hear your thoughts @jtramm as to whether that would cause problems elsewhere.

Reply:
It also better reflects the organization now, since the functions in this file are no longer responsible for managing separate streams.